Mimetypes

runcom opened the pull request:
These are some changes from https://github.com/projectatomic/skopeo/pull/62 that don't break skopeo in any way but lay the ground for future work (#62-like)

This depends on #61 and #65 (I'll rebase after those are merged)

@mtrmac PTAL

mtrmac commented on docker/utils/manifest.go:
```diff
@@ -10,11 +10,22 @@ import (
+		V2Schema2MIMEType,
+		V2ListMIMEType,
+	}
+}
```
Could just be a public `var`? OTOH that’s perhaps too risky.

Drop the `Get` prefix though :)

runcom replied:
> Drop the Get prefix though :)

ouch

mtrmac commented on docker/utils/manifest.go:
```diff
@@ -10,11 +10,22 @@ import (
+	// V2Schema2MIMEType MIME type represents Docker manifest schema 2
+	V2Schema2MIMEType = "application/vnd.docker.distribution.manifest.v2+json"
+	// V2ListMIMEType MIME type represents Docker *fat* manifest aka manifest list
+	V2ListMIMEType = "application/vnd.docker.distribution.manifest.list.v2+json"
```
Please keep the `Docker` naming here.

runcom replied:
I will, I forgot to put it back

mtrmac commented on cmd/skopeo/copy.go:
```diff
@@ -54,7 +54,7 @@ func copyHandler(context *cli.Context) {
 	signBy := context.String("sign-by")
 
-	manifest, err := src.GetManifest()
+	manifest, _, err := src.GetManifest()
```
(This is the right thing to do now.

Eventually we might need to do conversions, i.e. turn the core of `copy` into a function available in the library (which would be called for `pull` operation by k8s).  But that’s for much later, if ever.)

mtrmac commented on types/types.go:
```diff
@@ -37,8 +37,7 @@ type ImageSource interface {
 	// GetManifest returns the image's manifest.  It may use a remote (= slow) service.
-	// FIXME? This should also return a MIME type if known, to differentiate between schema versions.
-	GetManifest() ([]byte, error)
+	GetManifest() ([]byte, string, error)
```
Please update the comment also to say what the second parameter is, and that "" means “unknown”.

mtrmac commented:
I can’t see any dependency on #65 FWIW.

runcom commented:
> I can’t see any dependency on #65 FWIW.

I had included the commit here - so now it's merged and I've rebased.

Also, I've fixed what you requested - after #61 is merged I will rebase here and we could merge this also if we want

mtrmac commented:
Thanks, LGTM. Will merge after we agree on what we mean in #61 :)

runcom commented:
I'll reopen once I re-fix this
